src/components/Taxonomy/Taxonomy.tsx

```diff
@@ -113,13 +113,22 @@ const UserLabelForm = ({ onAddLabel, onFinish, path }: UserLabelFormProps) => {
   );
 };
 
-const SelectedList = ({ isReadonly }) => {
+const SelectedList = ({ isReadonly, flatItems } : { isReadonly:boolean, flatItems:TaxonomyItem[] }) => {
   const [selected, setSelected] = useContext(TaxonomySelectedContext);
   const { showFullPath, pathSeparator = " / " } = useContext(TaxonomyOptionsContext);
 
+  const selectedLabels = selected.map((selectedItem: string[]) =>
+    selectedItem.map(
+      (value: string) => {
+        const label = flatItems.find(taxonomyItem => taxonomyItem.path[taxonomyItem.path.length - 1] === value)?.label;
+        
+        return label ?? value;  
+      }),
+  );
+  
   return (
     <div className={styles.taxonomy__selected}>
-      {selected.map(path => (
+      {selectedLabels.map(path => (
```

Comment:
I'm curious what this does when @yyassi-heartex 's changes are in the mix? I'm thinking this should be tested in conjunction as well.

Reply:
me too but someone has to merge first

Reply:
agreed, if this merges first i can resolve any conflicts that come up in my branch - no biggie

Reply:
Brandon, that was a good catch about the chip deletion. I have updated to handle this too. ready to merge